tests/countries/test_tuvalu.py

```diff
@@ -0,0 +1,439 @@
+            "2018-03-31",
+            "2019-04-20",
+            "2020-04-11",
+            "2021-04-03",
+            "2022-04-16",
+            "2023-04-08",
+            "2024-03-30",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(1990, 2050))
+        dt = (
+            "1995-04-17",
+            "2006-04-17",
+            "2011-04-25",
+            "2017-04-17",
+            "2028-04-17",
+        )
+        self.assertHolidayName(f"{name} (fakamatakuga)", dt)
+
+    def test_easter_monday(self):
+        name = "Toe Tu aso gafua"
+        dt = (
+            "1999-04-05",
+            "2000-04-24",
+            "2010-04-05",
+            "2018-04-02",
+            "2019-04-22",
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(1990, 2050))
+        for holidays in self.subdiv_holidays.values():
+            self.assertHolidayName(name, holidays, dt)
+            self.assertHolidayName(name, holidays, range(1990, 2050))
+
+    def test_christmas_day(self):
+        name = "Kilisimasi"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(1990, 2050)))
+        for holidays in self.subdiv_holidays.values():
+            self.assertHolidayName(name, holidays, range(1990, 2050))
+
+    def test_all_holidays(self):
+        holidays_found = set()
+        for subdiv in Tuvalu.subdivisions:
+            holidays_found.update(
+                Tuvalu(
+                    subdiv=subdiv,
+                    observed=False,
+                    years=(1990, 2007, 2012, 2015, 2023),
+                ).values()
+            )
+        all_holidays = {
+            "Tausaga Fou",
+            "Aso toe tu",
+            "Aso Tapu",
+            "Toe Tu aso gafua",
+            "Aso Atefenua",
+            "Te Aso o te Tala Lei",
+            "Asofanau Fafine",
+            "Aso Tamaliki",
+            "Aso tupulaga",
+            "Tutokotasi",
+            "Aso fanau o te sui ote Tupu",
+            "Kilisimasi",
+            "Aso Faipele",
+            "Aso o te matagi",
+            "Te Aso o te Paula",
+            "Aho o te Fakavae",
+            "Te Po o Tefolaha",
+            "Po Lahi",
+            "Te Aso o te Setema",
+            "Bogin te Ieka",
+            "Te Aso O Tutasi",
+            "Aso o te Tala Lei",
+            "Te Aso Fiafia",
+        }
+        self.assertEqual(all_holidays, holidays_found)
+
+    def test_cyclone_day(self):
+        fun_holidays = Tuvalu(subdiv="FUN", years=range(1990, 2050))
+        name = "Aso o te matagi"
+        for subdiv, holidays in self.subdiv_holidays.items():
+            if subdiv == "FUN":
+                self.assertHolidayName(
+                    name, fun_holidays, (f"{year}-10-21" for year in range(1990, 2050))
+                )
```

Comment:
```suggestion
        name = "Aso o te matagi"
        for subdiv, holidays in self.subdiv_holidays.items():
            if subdiv == "FUN":
                self.assertHolidayName(
                    name, holidays, (f"{year}-10-21" for year in range(1990, 2050))
                )
            else:
                self.assertNoHolidayName(name, holidays)
```
And the same for the rest of subdivisions' holidays.

Reply:
```python3

            else:
                self.assertNoHolidayName(name, holidays)
```
this part throwes error:-
```bash

self = <tests.countries.test_tuvalu.TestTuvalu testMethod=test_gospel_day>

    def test_gospel_day(self):
        name = "Te Aso o te Tala Lei"
        dt = (
            "2001-05-14",
            "2002-05-13",
            "2003-05-12",
            "2004-05-10",
            "2005-05-09",
            "2006-05-15",
            "2007-05-14",
        )
        self.assertHolidayName(name, dt)
        self.assertHolidayName(name, range(1990, 2050))

        for subdiv, holidays in self.subdiv_holidays.items():
            if subdiv == "NKL":
                self.assertHolidayName(
                    name, holidays, (f"{year}-05-10" for year in range(1990, 2050))
                )
            else:
>               self.assertNoHolidayName(name, holidays)

tests\countries\test_tuvalu.py:75:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests\common.py:276: in assertNoHolidayName
    self._assertNoHolidayName(name, "holidays", *args)
tests\common.py:261: in _assertNoHolidayName
    self.assertFalse(holidays.get_named(name, lookup="exact"), name)
E   AssertionError: [datetime.date(2048, 5, 11), datetime.date(2049, 5, 10), datetime.date(1990, 5, 14), datetime.date(1991, 5, 13), datetime.date(1992, 5, 11), datetime.date(1993,
 5, 10), datetime.date(1994, 5, 9), datetime.date(1995, 5, 15), datetime.date(1996, 5, 13), datetime.date(1997, 5, 12), datetime.date(1998, 5, 11), datetime.date(1999, 5, 10), date
time.date(2000, 5, 15), datetime.date(2001, 5, 14), datetime.date(2002, 5, 13), datetime.date(2003, 5, 12), datetime.date(2004, 5, 10), datetime.date(2005, 5, 9), datetime.date(200
6, 5, 15), datetime.date(2007, 5, 14), datetime.date(2008, 5, 12), datetime.date(2009, 5, 11), datetime.date(2010, 5, 10), datetime.date(2011, 5, 9), datetime.date(2012, 5, 14), da
tetime.date(2013, 5, 13), datetime.date(2014, 5, 12), datetime.date(2015, 5, 11), datetime.date(2016, 5, 9), datetime.date(2017, 5, 15), datetime.date(2018, 5, 14), datetime.date(2
019, 5, 13), datetime.date(2020, 5, 11), datetime.date(2021, 5, 10), datetime.date(2022, 5, 9), datetime.date(2023, 5, 15), datetime.date(2024, 5, 13), datetime.date(2025, 5, 12), 
datetime.date(2026, 5, 11), datetime.date(2027, 5, 10), datetime.date(2028, 5, 15), datetime.date(2029, 5, 14), datetime.date(2030, 5, 13), datetime.date(2031, 5, 12), datetime.dat
e(2032, 5, 10), datetime.date(2033, 5, 9), datetime.date(2034, 5, 15), datetime.date(2035, 5, 14), datetime.date(2036, 5, 12), datetime.date(2037, 5, 11), datetime.date(2038, 5, 10
), datetime.date(2039, 5, 9), datetime.date(2040, 5, 14), datetime.date(2041, 5, 13), datetime.date(2042, 5, 12), datetime.date(2043, 5, 11), datetime.date(2044, 5, 9), datetime.date(2045, 5, 15), datetime.date(2046, 5, 14), datetime.date(2047, 5, 13)] is not false : Te Aso o te Tala Lei
============================================================================= short test summary info ============================================================================= 
FAILED tests/countries/test_tuvalu.py::TestTuvalu::test_gospel_day - AssertionError: [datetime.date(2048, 5, 11), datetime.date(2049, 5, 10), datetime.date(1990, 5, 14), datetime.date(1991, 5, 13), datetime.date(1992, 5, 11), datetime.date(1993...
========================================================================== 1 failed, 30 passed in 0.46s =========================================================================== 
```
What should be the workaround for the issues caused by this holiday

Reply:
> self.assertNoHolidayName(name, holidays)

Since this holiday (with this name) is also nation-wide, this check is not required.